mne_bids/tests/test_write.py

```diff
@@ -77,6 +79,56 @@ def _validate(output_path):
     return _validate
 
 
+@requires_version('pybv', '0.2.0')
+def _test_anonymize(raw, bids_basename, events_fname=None, event_id=None):
+    output_path = _TempDir()
+    write_raw_bids(raw, bids_basename, output_path,
+                   events_data=events_fname,
+                   event_id=event_id, anonymize=dict(daysback=33000),
+                   overwrite=False)
+    scans_tsv = make_bids_basename(
+        subject=subject_id, session=session_id, suffix='scans.tsv',
+        prefix=op.join(output_path, 'sub-01', 'ses-01'))
+    data = _from_tsv(scans_tsv)
+    assert datetime.strptime(data['acq_time'][0],
+                             '%Y-%m-%dT%H:%M:%S').year < 1925
+    return output_path
+
+
+def test_stamp_to_dt():
+    """Test conversions of meas_date to datetime objects."""
+    meas_date = (1346981585, 835782)
+    meas_datetime = _stamp_to_dt(meas_date)
+    assert(meas_datetime == datetime(2012, 9, 7, 1, 33, 5, 835782,
+                                     tzinfo=timezone.utc))
+    meas_date = (1346981585,)
+    meas_datetime = _stamp_to_dt(meas_date)
+    assert(meas_datetime == datetime(2012, 9, 7, 1, 33, 5, 0,
+                                     tzinfo=timezone.utc))
+
+
+def test_get_anonymization_daysback():
+    """Test daysback querying for anonymization."""
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    raw = mne.io.read_raw_fif(raw_fname)
+    daysback_min, daysback_max = _get_anonymization_daysback(raw)
+    # max_val off by 1 on Windows for some reason
+    assert abs(daysback_min - 28461) < 2 and abs(daysback_max - 36880) < 2
+    raw2 = raw.copy()
+    raw2.info['meas_date'] = (np.int32(1158942080), np.int32(720100))
+    raw3 = raw.copy()
+    raw3.info['meas_date'] = (np.int32(914992080), np.int32(720100))
+    daysback_min, daysback_max = get_anonymization_daysback([raw, raw2, raw3])
+    assert abs(daysback_min - 29850) < 2 and abs(daysback_max - 35446) < 2
+    raw4 = raw.copy()
+    raw4.info['meas_date'] = (np.int32(4992080), np.int32(720100))
```

Comment:
can you set `raw.info['meas_date'] = None` and see what it gives?

basically make sure that `meas_date=None` does not break anything once @bloyl 's PR is in.

Reply:
I need the dates to be in the 2000s for the tests to use the same number of daysback for each test. Otherwise there's just more magic numbers for daysback.

Reply:
not sure what you mean. This function shouldn't break for dates before 2000 and for `meas_date=None`. It should either give explicit error or give something reasonable.

Reply:
So if meas_date gets set to None, then the date gets set to 1970, which needs a different daysback argument than for the rest of the test files which were collected in the 2000s. Unless Luke set it to 2000 by default.

Reply:
In the new PR - If `raw.info['meas_date']` is `None` then it stays `None`.

When this gets written to a fif file it is converted to a tuple (0, 2^31-1) - https://github.com/mne-tools/mne-python/blob/c80badd8939423d9c5dd342a60135bd8ead6cc13/mne/io/write.py#L20-L24.

then when being read back in it gets converted back to `None`
https://github.com/mne-tools/mne-python/blob/c80badd8939423d9c5dd342a60135bd8ead6cc13/mne/io/meas_info.py#L1205-L1209

Not sure if this helps this conversation or not?

Reply:
Thanks @bloyl! That ought to do it, I will just delete the magic numbers and allow meas_date=None to pass the test as well

Reply:
This is not yet resolved. Does `get_anonymization_daysback` work if one of the `raw` objects has `raw.info['meas_date'] = None`? This does not depend on Luke's PR as far as I understand because here you just manipulate datetime objects to get the right range of dates.

Reply:
Hmmm it depends on how it's handled when daysback is not None to anonymize_info and meas_date is None in Luke's PR, I'll test that now.

Reply:
It says "Input info has 'meas_date' set to None. Removing all information from time/date structures. *NOT* performing any time shifts" and then as advertised 'meas_date' is None. Thus I think None meas_dates should be ignored. It doesn't really make sense to give them a fake date and time-shift them to me at least. Fixing now.